ga4gh/cli.py

```diff
@@ -1510,448 +1521,595 @@ def configtest_main(parser=None):
-
-    def run(self):
-        self.repoManager.check(self.doConsistencyCheck)
-
-
-class ListRunner(AbstractRepoCommandRunner):
-
-    def __init__(self, args):
-        super(ListRunner, self).__init__(args)
-
-    def run(self):
-        self.repoManager.list()
-
-
-class DestroyRunner(AbstractRepoCommandRunner):
-
-    def __init__(self, args):
-        super(DestroyRunner, self).__init__(args)
-
-    def run(self):
-        def func(): self.repoManager.destroy()
-        self.confirmRun(func, 'the Data Repository')
-
-
-class InitRunner(AbstractRepoCommandRunner):
-
-    def __init__(self, args):
-        super(InitRunner, self).__init__(args)
-
-    def run(self):
-        self.repoManager.init()
-
-
-class AddDatasetRunner(AbstractRepoDatasetCommandRunner):
-
-    def __init__(self, args):
-        super(AddDatasetRunner, self).__init__(args)
-
-    def run(self):
-        self.repoManager.addDataset(self.datasetName)
-
-
-class AddOntologyMapRunner(AbstractRepoAddMoveCommandRunner):
-
-    def __init__(self, args):
-        super(AddOntologyMapRunner, self).__init__(args)
-
-    def run(self):
-        self.repoManager.addOntologyMap(self.filePath, self.moveMode)
-
-
-class RemoveOntologyMapRunner(AbstractRepoCommandRunner):
-
-    def __init__(self, args):
-        super(RemoveOntologyMapRunner, self).__init__(args)
-        self.ontologyMapName = args.ontologyMapName
-
-    def run(self):
-        def func(): self.repoManager.removeOntologyMap(self.ontologyMapName)
-        self.confirmRun(func, 'ontology map {}'.format(self.ontologyMapName))
-
-
-class RemoveDatasetRunner(AbstractRepoDatasetCommandRunner):
-
-    def __init__(self, args):
-        super(RemoveDatasetRunner, self).__init__(args)
-
-    def run(self):
-        def func(): self.repoManager.removeDataset(self.datasetName)
-        self.confirmRun(func, 'dataset {}'.format(self.datasetName))
-
-
-class AddReferenceSetRunner(AbstractRepoAddMoveCommandRunner):
-
-    def __init__(self, args):
-        super(AddReferenceSetRunner, self).__init__(args)
-        self.metadata = {
-            'description': args.description,
-        }
-
-    def run(self):
-        self.repoManager.addReferenceSet(
-            self.filePath, self.moveMode, self.metadata)
+    def list(self):
+        """
+        Lists the contents of this repo.
+        """
+        self._openRepo()
+        # TODO this is _very_ crude. We need much more options and detail here.
+        self._repo.printSummary()
 
+    def verify(self):
+        """
+        Checks that the data pointed to in the repository works and
+        we don't have any broken URLs, missing files, etc.
+        """
+        self._openRepo()
+        self._repo.verify()
 
-class RemoveReferenceSetRunner(AbstractRepoCommandRunner):
+    def addOntology(self):
+        """
+        Adds a new Ontology to this repo.
+        """
+        self._openRepo()
+        name = self._args.name
+        if name is None:
```

Comment:
For all the types it appears there is no catch for blank name. If it's intended to allow for this it's fine, but I noticed there are checks for empty string in the getNameFromPath so I wanted to make sure this is the desired functionality.

`python repo_dev.py add-readgroupset l a ga4gh-example-data/HG00096.bam --name ""`